File: fams-bo/resource/com/purefun/fams/core/bo/otw/TestBO2_OTW.java
package com.purefun.fams.core.bo.otw;

import com.purefun.fams.core.bo.TestBO2;
import com.purefun.fams.core.bo.pro.TestBO2_PRO;
import com.google.protobuf.InvalidProtocolBufferException;
import com.purefun.fams.core.bo.commom.ICommon_OTW;
import com.google.protobuf.Any;

public class TestBO2_OTW implements ICommon_OTW {
    TestBO2_PRO.TestBO2.Builder builder = null;
    TestBO2 bo = null;

    public TestBO2_OTW() {
        builder = TestBO2_PRO.TestBO2.newBuilder();
        bo= new TestBO2();
        builder.setUuid(bo.uuid);
        builder.setBoid(bo.boid);
        builder.setDestination(bo.destination);
    }

    public TestBO2_OTW(byte[] message) throws InvalidProtocolBufferException {
        builder = TestBO2_PRO.TestBO2.newBuilder();
        bo= new TestBO2();
        TestBO2_PRO.TestBO2 receive = TestBO2_PRO.TestBO2.parseFrom(message);
        setWorkid(receive.getWorkid());
        setHomeaddress(receive.getHomeaddress());
        setUuid(receive.getUuid());
        setBoid(receive.getBoid());
        setDestination(receive.getDestination());
    }

    public TestBO2_OTW(TestBO2 bofrom){
        builder = TestBO2_PRO.TestBO2.newBuilder();
        bo= new TestBO2();
        setWorkid(bofrom.workid);
        setHomeaddress(bofrom.homeaddress);
        setUuid(bofrom.uuid);
        setBoid(bofrom.boid);
        setDestination(bofrom.destination);
    }

    public byte[] serial() {
        return builder.build().toByteArray();
    }

    @Override
    public com.google.protobuf.GeneratedMessageV3.Builder getBuilder() { 
        return builder;
    }

    @Override
    public TestBO2 getBo() { 
        return bo;
    }

    public java.lang.String getWorkid() {
        return builder.getWorkid();
    }

    public void setWorkid(java.lang.String workid) {
        bo.workid = workid;
        builder.setWorkid(workid);
    }

    public java.lang.String getHomeaddress() {
        return builder.getHomeaddress();
    }

    public void setHomeaddress(java.lang.String homeaddress) {
        bo.homeaddress = homeaddress;
        builder.setHomeaddress(homeaddress);
    }

    public java.lang.String getUuid() {
        return builder.getUuid();
    }

    public void setUuid(java.lang.String uuid) {
        bo.uuid = uuid;
        builder.setUuid(uuid);
    }

    public long getBoid() {
        return builder.getBoid();
    }

    public void setBoid(long boid) {
        bo.boid = boid;
        builder.setBoid(boid);
    }

    public java.lang.String getDestination() {
        return builder.getDestination();
    }

    public void setDestination(java.lang.String destination) {
        bo.destination = destination;
        builder.setDestination(destination);
    }

    public String toString() {
        return "TestBO2_OTW ["+
            "uuid = " + getUuid() +"," +
            "boid = " + getBoid() +"," +
            "destination = " + getDestination() +"," +
            "workid = " + getWorkid() +"," +
            "homeaddress = " + getHomeaddress() +"," +
         "]";
    }
}

File: fams-python-service/src/com/purefun/fams/core/bo/otw/TestBO2_OTW.py
from src.core.common.ICommon_OTW import ICommon_OTW
from src.com.purefun.fams.core.bo.pro import TestBO2_pb2
from src.com.purefun.fams.core.bo.model.TestBO2 import TestBO2

class TestBO2_OTW(ICommon_OTW):

    def __init__(self, byteMsg = None):
        self._bo_pro = TestBO2_pb2.TestBO2()
        self._bo = TestBO2()

        if byteMsg is not None:
            self._bo_pro.ParseFromString(byteMsg)
            self.__setDataFromPB()
        else:
            self.__setDataFromBO()

    def __setDataFromBO(self):
        self._bo_pro.workid = self._bo.workid
        self._bo_pro.homeaddress = self._bo.homeaddress
        self._bo_pro.uuid = self._bo.uuid
        self._bo_pro.boid = self._bo.boid
        self._bo_pro.destination = self._bo.destination

    def __setDataFromPB(self):
        self._bo.workid = self._bo_pro.workid
        self._bo.homeaddress = self._bo_pro.homeaddress
        self._bo.uuid = self._bo_pro.uuid
        self._bo.boid = self._bo_pro.boid
        self._bo.destination = self._bo_pro.destination

    def getBO(self):
        return self._bo

    def getProBO(self):
        return self._bo_pro

    def getWorkid(self):
        return self._bo.workid

    def setWorkid(self, workid):
        self._bo.workid = workid
        self._bo_pro.workid = workid

    def getHomeaddress(self):
        return self._bo.homeaddress

    def setHomeaddress(self, homeaddress):
        self._bo.homeaddress = homeaddress
        self._bo_pro.homeaddress = homeaddress

    def getUuid(self):
        return self._bo.uuid

    def setUuid(self, uuid):
        self._bo.uuid = uuid
        self._bo_pro.uuid = uuid

    def getBoid(self):
        return self._bo.boid

    def setBoid(self, boid):
        self._bo.boid = boid
        self._bo_pro.boid = boid

    def getDestination(self):
        return self._bo.destination

    def setDestination(self, destination):
        self._bo.destination = destination
        self._bo_pro.destination = destination

    def toString(self):
        return "TestBO2_OTW ["+"workid = " + self.getWorkid() +"," +"homeaddress = " + self.getHomeaddress() +"," +"uuid = " + self.getUuid() +"," +"boid = " + str(self.getBoid()) +"," +"destination = " + self.getDestination() +"," +"]"
